src/components/unique-token/UniqueTokenImage.tsx

```diff
@@ -24,84 +22,103 @@ function getFallbackTextColor(bg: string, isDarkMode: boolean, colors: Colors) {
+        collectionName,
+        tokenId,
+        name,
+        uniqueId,
+      })}
+    </Text>
+  );
+});
+
 export const UniqueTokenImage = React.memo(function UniqueTokenImage({
-  backgroundColor: givenBackgroundColor,
+  backgroundColor,
   imageUrl,
   lowResImageUrl,
   collectionName,
   name,
   uniqueId,
-  fullUniqueId,
   id,
-  address,
   isCard = false,
-  mimeType,
-  transformSvgs = true,
+  optimisticImageLoading = false,
 }: UniqueTokenImageProps) {
   const { isDarkMode } = useColorMode();
   const { colors } = useTheme();
   const { hiddenTokens } = useHiddenTokens();
 
+  const [isLoading, setIsLoading] = useState(optimisticImageLoading);
   const [errorLoadingImage, setErrorLoadingImage] = useState(false);
-  const [isLoaded, setIsLoaded] = useState(false);
 
-  const onLoad = useCallback(() => setIsLoaded(true), [setIsLoaded]);
-  const onError = useCallback(() => setErrorLoadingImage(true), [setErrorLoadingImage]);
+  const onLoad = useCallback(() => setIsLoading(false), [setIsLoading]);
+  const onError = useCallback(
+    (...args: unknown[]) => {
+      logger.warn(`[UniqueTokenImage] Error loading image: ${args} for ${uniqueId}`);
+      setErrorLoadingImage(true);
+    },
+    [setErrorLoadingImage, uniqueId]
+  );
 
   const isHiddenToken = useMemo(() => {
-    return hiddenTokens.find(token => token === fullUniqueId);
-  }, [hiddenTokens, fullUniqueId]);
+    return hiddenTokens.find(token => isLowerCaseMatch(token, uniqueId));
+  }, [hiddenTokens, uniqueId]);
 
-  const backgroundColor = givenBackgroundColor;
-  const isENS = address?.toLowerCase() === ENS_NFT_CONTRACT_ADDRESS;
-  const isSVG = mimeType === 'image/svg+xml';
   const hasImage = imageUrl !== null && imageUrl !== undefined;
-
-  const shouldShowSvg = hasImage && isSVG && !errorLoadingImage && !transformSvgs;
-  const shouldShowRegularImage = hasImage && !isSVG && !errorLoadingImage;
-  const shouldShowTextFallback = (!shouldShowSvg && !shouldShowRegularImage) || (isHiddenToken && isCard);
+  const shouldShowRegularImage = hasImage && !errorLoadingImage;
+  const shouldShowTextFallback = !shouldShowRegularImage || (isHiddenToken && isCard);
 
   return (
     <Centered backgroundColor={backgroundColor} style={StyleSheet.absoluteFill}>
-      {shouldShowSvg && (
-        <RemoteSvg
-          fallbackIfNonAnimated={!isENS || isCard}
-          fallbackUri={svgToPngIfNeeded(imageUrl, true)}
-          lowResFallbackUri={imageUrl}
-          onError={onError}
-          style={StyleSheet.absoluteFill}
-          uri={imageUrl}
-        />
-      )}
       {shouldShowRegularImage && (
         <>
           <RainbowImage onError={onError} onSuccess={onLoad} source={{ url: imageUrl }} style={StyleSheet.absoluteFillObject} />
-          {!isLoaded && lowResImageUrl && <RainbowImage source={{ url: lowResImageUrl }} style={StyleSheet.absoluteFillObject} />}
+          {optimisticImageLoading && isLoading && lowResImageUrl && (
```

Comment:
if lowResImage url is empty string this can error, wrap in Boolean() if so

Reply:
I thought an empty string was falsy?

Reply:
I've used this pattern so much and never noticed that. Would love to sync on this on Monday if you're able to! Really intrigued.